crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -296,6 +302,16 @@ impl SimpleFaucet {
                     "Failed to execute PaySui transactions in faucet after {elapsed}. Coin will \
                      not be reused."
                 );
+
+                // We set the inflight status to false so that the async thread that
+                // retries this transactions will attempt to try again.
+                if let Err(err) = self.wal.lock().await.set_in_flight(coin_id, false) {
+                    error!(
+                        ?coin_id,
```

Comment:
Can you include the `uuid` in the log as well, (and the `recipient`), like in the warning above?  That way we get more data points to trace a common thread for a given request.